torch/library.py

```diff
@@ -954,6 +954,118 @@ def register(func):
         return register(func)
 
 
+def register_vmap(
+    op: _op_identifier,
+    func: Callable,
+    /,
+    *,
+    lib=None,
+) -> None:
+    r"""Register a vmap implementation to support :func:`torch.vmap` for this custom op.
+
+    In order for an operator to work with :func:`torch.vmap`, you may need to register a
+    vmap implementation in the following signature:
+
+        ``vmap_func(info, in_dims: Tuple[Optional[int]], *args, **kwargs)``,
+
+    where ``*args`` and ``**kwargs`` are the arguments and kwargs for ``op``.
```

Comment:
just to check: we do not support Tensor args in kwargs, right?

Reply:
I added a check now.